src/engraving/dom/note.cpp

```diff
@@ -2262,6 +2262,9 @@ void Note::setTrack(track_idx_t val)
             seg->setTrack(val);
         }
     }
+    if (m_tieBack) {
+        m_tieBack->setTrack2(val);
```

Comment:
What is the difference between setTrack and setTrack2?

Reply:
It's for cross-staff notation. Most spanners have the ability to cross from one staff to another, so `m_track` is the starting track of the spanner, and `m_track2` is the ending track.
![image](https://github.com/user-attachments/assets/b6df3e75-11ea-419a-8309-bea46eca1718)
In this case, when staves get deleted, we need to update all the tracks but we were forgetting to set `track2`, which was causing the crash.